src/App.tsx

```diff
@@ -8,37 +8,119 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { client } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
Add a generic types everywher
```suggestion
  const [errorMessage, setErrorMessage] = useState<string>('');
```